server/xpub-server/entities/manuscript/resolvers/updateManuscript.js

```diff
@@ -1,10 +1,33 @@
+const Joi = require('joi')
 const pug = require('pug')
+const dns = require('dns')
+const util = require('util')
 const config = require('config')
 const mailer = require('@pubsweet/component-send-email')
 const logger = require('@pubsweet/logger')
-
 const { Manuscript, User } = require('@elifesciences/xpub-model')
 
+const resolveMx = util.promisify(dns.resolveMx)
+
+const isValidEmail = async email => {
+  let result = false
+  try {
+    await Joi.validate(
+      email,
+      Joi.string()
+        .email()
+        .required(),
+    )
+    const tld = email.split('@')[1]
+    const lookup = await resolveMx(tld)
```

Comment:
taking the opportunity to ask whether all the `await` constructs will result in a blocked process (iowait) or the single Node process will switch to some other thing from the event queue? I think the latter but better to get confirmation given all the network calls going on

Reply:
separately: does this need to be stubbed in the context of unit tests (no network, run on a train) or disabled via configuration?

Reply:
@giorgiosironi Node will start process something else from the event queue during an `await`

Reply:
@giorgiosironi / @damnhipster .. the notification is now done asynchronously with the new commit